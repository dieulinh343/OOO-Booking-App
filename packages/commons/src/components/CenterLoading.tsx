import React from 'react';
import { Loader } from '@ahaui/react';

export function CenterLoading ({
  hideText,
  text,
  duration,
  size = 'large',
}: {
  hideText: boolean;
  text: string;
  duration: number;
  size?: string;
}) {
  return (
    <div className="u-positionAbsolute u-positionFull u-flex u-alignContentCenter u-justifyContentCenter">
      <div className="u-flex u-alignItemsCenter u-justifyContentCenter u-flexColumn">
        <div className="u-flexShrink-0">
          <Loader size={size} duration={duration} />
        </div>
        {!hideText && <div className="u-textLight u-marginTopMedium">{text || 'The system is processing. Please waitâ€¦'}</div>}
      </div>
    </div>
  );
}

export default CenterLoading;
