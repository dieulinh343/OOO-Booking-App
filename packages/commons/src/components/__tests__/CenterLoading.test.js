import React from 'react';
import { shallow } from 'enzyme';
import { CenterLoading } from '../CenterLoading';

describe('components/CenterLoading', () => {
  let wrapper;
  let props;

  beforeEach(() => {
    wrapper = shallow(<CenterLoading {...props} />);
  });

  it('should render successfully', () => {
    expect(wrapper.length).toEqual(1);
    expect(wrapper).toMatchSnapshot();
  });
});
