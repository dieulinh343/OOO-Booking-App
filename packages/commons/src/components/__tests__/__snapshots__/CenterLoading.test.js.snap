// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/CenterLoading should render successfully 1`] = `
<div
  className="u-positionAbsolute u-positionFull u-flex u-alignContentCenter u-justifyContentCenter"
>
  <div
    className="u-flex u-alignItemsCenter u-justifyContentCenter u-flexColumn"
  >
    <div
      className="u-flexShrink-0"
    >
      <Loader
        duration={2000}
        size="large"
      />
    </div>
    <div
      className="u-textLight u-marginTopMedium"
    >
      The system is processing. Please waitâ€¦
    </div>
  </div>
</div>
`;
